src/libopensc/reader-pcsc.c

```diff
@@ -264,8 +264,10 @@ static int pcsc_internal_transmit(sc_reader_t *reader,
 				  recvbuf, &dwRecvLength);
 		}
 		else {
-			rv = priv->gpriv->SCardControl(card, (DWORD) control, sendbuf, dwSendLength,
-				  recvbuf, dwRecvLength, &dwRecvLength);
+			if (priv->gpriv->SCardControl != NULL) {
+				rv = priv->gpriv->SCardControl(card, (DWORD) control, sendbuf, dwSendLength,
+						recvbuf, dwRecvLength, &dwRecvLength);
+			}
```

Comment:
nit: I would put it in `else if` branch here to avoid too much indentation:
```
		} else if (priv->gpriv->SCardControl != NULL) {
			rv = priv->gpriv->SCardControl(card, (DWORD) control, sendbuf, dwSendLength,
					recvbuf, dwRecvLength, &dwRecvLength);
		}
```